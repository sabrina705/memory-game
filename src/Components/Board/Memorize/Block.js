import React from 'react';
import classnames from 'classnames';

const Block = ({ gem, width }) =>
  <div className={classnames('block', { gem })} style={{ width }} />;

export default Block;
