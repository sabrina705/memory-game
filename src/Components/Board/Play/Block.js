import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';

export class Block extends Component {
  static propTypes = {
    width: PropTypes.number.isRequired,
    index: PropTypes.number.isRequired,
    onClick: PropTypes.func.isRequired,
    selected: PropTypes.bool.isRequired,
    gem: PropTypes.bool.isRequired,
  };

  handleClick = () => {
    this.props.onClick(this.props.index);
  };

  render() {
    const { selected, width, gem } = this.props;
    const classes = classnames('block', { gem, selected });

    return (
      <div style={{ width }} onClick={this.handleClick} className={classes} />
    );
  }
}

export default Block;
