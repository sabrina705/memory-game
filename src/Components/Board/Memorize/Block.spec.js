import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Block from './Block';

it('Memorize/Block renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Block gem={true} width={50} />, div);
});

it('Memorize/Block matches snapshot gem true', () => {
  const component = renderer.create(
    <Block gem={true} width={50} />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('Memorize/Block matches snapshot gem false', () => {
  const component = renderer.create(
    <Block gem={false} width={50} />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
