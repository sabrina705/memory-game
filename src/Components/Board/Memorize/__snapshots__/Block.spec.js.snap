exports[`test Memorize/Block matches snapshot gem false 1`] = `
<div
  className="block"
  style={
    Object {
      "width": 50,
    }
  } />
`;

exports[`test Memorize/Block matches snapshot gem true 1`] = `
<div
  className="block gem"
  style={
    Object {
      "width": 50,
    }
  } />
`;
